packages/@sfx/sayt-driver-plugin/test/unit/browser/sayt-driver-plugin.test.ts

```diff
@@ -0,0 +1,193 @@
+
+    it('should have two dependencies: "dom_events" and "sayt"', () => {
+      expect(Driver.metadata.depends).to.deep.equal(['dom_events', 'sayt']);
+    });
+  });
+
+  describe('.register()', () => {
+    beforeEach(() => {
+      Driver = {};
+    })
+
+    it('should keep a reference to all registered plugins', () => {
+      const Plugins = {
+        a: () => 'a',
+        b: () => true,
+      }
+      Driver = new SaytDriverPlugin();
+
+      expect(Driver.core).to.be.undefined;
+      Driver.register(Plugins);
+
+      expect(Driver.core).to.deep.equal(Plugins);
+    });
+  });
+
+  describe('.ready()', () => {
+    let registerListener: any;
+
+    beforeEach(() => {
+      registerListener = spy(Driver.core['dom_events'], 'registerListener');
+    });
+
+    it('should register an event listener for receiving sayt API requests', () => {
+      Driver.ready();
+
+      registerListener.getCall(0).calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.unregister()', () => {
+    let unregisterListener;
+
+    beforeEach(() => {
+      unregisterListener = spy(Driver.core['dom_events'], 'unregisterListener');
+    });
+
+    it('should unregister the sayt event listener', () => {
+      Driver.unregister();
+
+      expect(unregisterListener).to.have.been.calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.autocompleteCallback', () => {
+    it('should return an array of search term strings from the sayt response', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' , notValue: 'z' },
+            { value: 'b' },
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'b', 'c']);
+    });
+
+    it('should only return entries that have a searchterm value', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' },
+            {},
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'c']);
+    });
+
+    it('should return an empty array if there are no searchTerms', () => {
+      const response = {
+        result: {
+          searchTerms: [],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal([]);
+    });
+  });
+
+  describe('.sendSaytAPIRequest()', () => {
+    const saytDataPayload = {
+      query: 'shirt',
+      collection: 'backup',
+    };
+    let autocompleteCallback;
+
+    beforeEach(() => {
+      autocompleteCallback = stub(Driver, 'autocompleteCallback');
+    });
+
+    it('should make a search call through the sayt client', () => {
+      const autocomplete = spy(Driver.core.sayt, 'autocomplete');
```

Comment:
Use a stub. Ditto `driver.core` comment. If you need the stub to return a promise, you can look at [`stub.resolves`](https://sinonjs.org/releases/latest/stubs/#stubresolvesvalue).